transcom/mymove
# B:18828 - Add Create Okta Account Logic to UpdateRequestedOfficeUser

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-18828)

## Summary

MERGE AFTER B-18821
MERGE AFTER B-18825

This PR introduces additions to the UpdateRequestedOfficeUserHandler. These additions includes the following:
1. Create okta Account Logic - (This only happens in the case that param.Status == "APPROVED")
2. UpdateRequestedOfficeUser will always run in all cases EXCEPT the case when CreateAccountOkta logic runs and fails.

[Okta API Specifications](https://developer.okta.com/docs/reference/api/roles/) explains more about the okta api specifications

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Start the client & server and log in as a MM Admin
2. Using postman or swagger UI
3. Send a PATCH request to /requested-office-users/{officeUserId} (Make sure Status is set to "APPROVED")
4. Provide relevant needed data for Okta account creation such as otherUniqueId, Email, Firstname, Lastname, gsaID, EDIPI, MobilePhone

Example: May need to adjust for your local env
{
  "firstName": "Leo",
  "middleInitials": "J",
  "lastName": "Spaceman",
  "email": "user5435@userdomain.com",
  "telephone": "212-555-5555",
  "roles": [
     {
      "name": "Transportation Invoicing Officer",
      "roleType": "transportation_invoicing_officer"
    }
  ],
  "transportationOfficeId": "f344dde8-1c04-4a37-b66d-3f1bab7c494c",
  "edipi": "1111111110",
  "otherUniqueId": "1111111110",
  "rejectionReason": "string",
  "status": "APPROVED"
}


### Backend


### Database

#### Any new migrations/schema changes:
